src/library/Tabs/Tabs.js

```diff
@@ -0,0 +1,295 @@
+/* @flow */
+import React, { cloneElement, Component } from 'react';
+import { canUseDOM } from 'exenv';
+import { createStyledComponent } from '../styles';
+import { withTheme } from '../themes';
+import { generateId } from '../utils';
+import { findAllByType } from '../utils/children';
+import Tab from './Tab';
+import TabList from './TabList';
+import TabPanel from './TabPanel';
+
+type Props = {
+  /**  Horizontal or vertical alignment of Tabs in the tab list */
+  align?: 'start' | 'center' | 'end' | 'justify',
+  /** @Private */
+  'aria-labelledby'?: string,
+  /** Content of Tabs; must be Tab components */
+  children?: React$Node,
+  /**
+   * Index of the selected Tab; primarily for use with uncontrolled components
+   */
+  defaultSelectedTabIndex?: number,
+  /** Id of Tabs */
+  id?: string,
+  /** Accessible label for Tabs */
+  label: string,
+  /** Maximum height of Tabs */
+  maxHeight?: number | string,
+  /** Maximum width of each Tab */
+  maxTabWidth?: number | string,
+  /** Called when a Tab is selected */
+  onChange?: (
+    selectedTabIndex: number,
+    event: SyntheticEvent<HTMLAnchorElement>
+  ) => void,
+  /** Position of the tab list in relation to the tab panel */
+  position?: 'bottom' | 'end' | 'start' | 'top',
+  /**
+   * Index of the selected Tab; primarily for use with controlled components.
+   * If this prop is specified, an `onChange` handler must also be specified.
+   * See also: `defaultSelectedTabIndex`
+   */
+  selectedTabIndex?: number,
+  /** @Private Theme */
+  theme: Object
+};
+
+type State = {
+  selectedTabIndex: number
+};
+
+type AnchorEvent = SyntheticEvent<HTMLAnchorElement>;
+
+const Root = createStyledComponent(
+  'div',
+  ({ maxHeight, position }) => {
+    const flexDirection = {
+      bottom: 'column-reverse',
+      end: 'row-reverse',
+      start: 'row',
+      top: 'column'
+    };
+
+    return {
+      display: 'flex',
+      flexDirection: flexDirection[position],
+      maxHeight
+    };
+  },
+  {
+    displayName: 'Tabs',
+    includeStyleReset: true
+  }
+);
+
+/**
+ * Tabs provide easy management for viewing related content in the same layout
+ * region.
+ */
+class Tabs extends Component<Props, State> {
+  static defaultProps = {
+    align: 'start',
+    maxTabWidth: '8.5em',
+    position: 'top'
+  };
+
+  state = {
+    selectedTabIndex: this.props.defaultSelectedTabIndex || 0
+  };
+
+  disabledTabIndexes: Set<number> = new Set();
+
+  id: string = this.props.id || `tabs-${generateId()}`;
+
+  lastIndex: number;
+
+  root: HTMLElement;
```

Comment:
Again, I believe that these refs generally need to be `?HTMLElement;`, as they are `undefined` prior to initial render and when the component has unmounted.